extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala

```diff
@@ -280,6 +284,38 @@ trait LineageParser {
       case p if p.nodeName == "SaveIntoDataSourceCommand" =>
         extractColumnsLineage(getQuery(plan), parentColumnsLineage)
 
+      case p
+          if p.nodeName == "AppendData"
+            || p.nodeName == "OverwriteByExpression"
+            || p.nodeName == "OverwritePartitionsDynamic" =>
+        val table = getPlanField[NamedRelation]("table", plan).name
+        extractColumnsLineage(getQuery(plan), parentColumnsLineage).map { case (k, v) =>
+          k.withName(s"$table.${k.name}") -> v
+        }
+
+      case p if p.nodeName == "MergeIntoTable" =>
```

Comment:
shall we handle `DeleteFromTable` ?

Reply:
I think `DeleteFromTable` corresponding to the lineage is a bit ambiguous and not considered for implementation.

Reply:
May be we can improve the semantics of delete operation together. e.g. `drop database`, `drop table`.. Do you think it is valuable ? we can create a issue first.

Reply:
OK, I create the #4049 to collect some suggestions in real business scenarios.